flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -197,12 +199,42 @@ public DynamicTableSink createDynamicTableSink(Context context) {
 
     // ~ Tools ------------------------------------------------------------------
 
-    private static LogStoreTableFactory createLogStoreTableFactory(Context context) {
-        return discoverLogStoreFactory(
-                context.getClassLoader(),
-                context.getCatalogTable()
-                        .getOptions()
-                        .getOrDefault(LOG_SYSTEM.key(), LOG_SYSTEM.defaultValue()));
+    private static Optional<LogStoreTableFactory> createOptionalLogStoreFactory(Context context) {
+        Configuration options = new Configuration();
+        context.getCatalogTable().getOptions().forEach(options::setString);
+
+        if (!options.get(CHANGE_TRACKING)) {
+            return Optional.empty();
+        }
+
+        if (options.get(LOG_SYSTEM) == null) {
+            // Use file store continuous reading
```

Comment:
Correct me if I'm wrong, the comment is not that accurate, should be `// Use file store reading` because the execution mode might be batch.

Reply:
Yes, but this is in the `createLogFactory` context, here it is implied that you want to create a reader for streaming, OK, I told you there is no LogSystem, you can only use file store continuous reading